students/chelsea_nayan/lesson06/expander.py

```diff
@@ -0,0 +1,42 @@
+'''Generates 1 million entries formatted like the exercise.csv'''
+
+# Checklist:
+# (1) Takes the data file and expands it to 10^6 entries
+# (2) Generage a guid in each column
+
+import csv
+import random
+from datetime import date
+from  uuid import uuid4
+
+# This is not a UUID ):
+# def rand_string_generator():
+#     '''Generates a random string of length using a-f and 0-9 alphanum characters'''
+#     rand_string = ''.join([random.choice('abcdef' + string.digits) for i in range(8)] + ['-'] +
+#                           [random.choice('abcdef' + string.digits) for i in range(4)] + ['-'] +
+#                           [random.choice('abcdef' + string.digits) for i in range(4)] + ['-'] +
+#                           [random.choice('abcdef' + string.digits) for i in range(4)] + ['-'] +
+#                           [random.choice('abcdef' + string.digits) for i in range(12)])
+#     return rand_string
+
+def rand_date_generator():
+    '''Generates a random date from the 1/01/10 to today'''
+    start = date(2010, 1, 1)  # (Year, Month, Day)
+    end = date.today()
+    rand_date = start + (end - start) * random.random()
+    return rand_date.strftime('%m/%d/%y')
```

Comment:
The way this one is coded, the year will be returned as only two digits, like this:

`75009197-d92c-4993-bfcc-2655f03413a9,2,3,4,5,02/02/19,`

This will cause both scripts to return 0 for all years and matches of "ao". Try changing the output format used by _strftime_ to make it return a 4-digit year and rerun your timing comparisons.